All OptimizeIndexJob are timing out after 1m (instead of 1h)

<!--- Provide a general summary of the issue in the Title above -->

## Environment & Context
<!--- Include as many relevant details about the environment you experienced the bug in -->

* Graylog Version: 4.0.2
* Elasticsearch Version: 7.10.2 (OSS), recently upgraded from 6 (DEB packages)
* Operating System: Ubuntu Server 20.04

## The problem

Shortly after upgrading GL from 4.0.1 to 4.0.2 and **ES from 6 to 7.10.2 (OSS)**, all new OptimizeIndexJob are failing with the following error in the logs (timeout after 60 seconds).

```
2021-02-12T11:44:40.808+01:00 ERROR [SystemJobManager] Unhandled error while running SystemJob <07c50bec-6d1f-11eb-b130-005056afb272> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob]
org.graylog.shaded.elasticsearch7.org.elasticsearch.ElasticsearchException: An error occurred: 
        at org.graylog.storage.elasticsearch7.ElasticsearchClient.exceptionFrom(ElasticsearchClient.java:134) ~[?:?]
        at org.graylog.storage.elasticsearch7.ElasticsearchClient.execute(ElasticsearchClient.java:97) ~[?:?]
        at org.graylog.storage.elasticsearch7.ElasticsearchClient.execute(ElasticsearchClient.java:90) ~[?:?]
        at org.graylog.storage.elasticsearch7.IndicesAdapterES7.optimizeIndex(IndicesAdapterES7.java:394) ~[?:?]
        at org.graylog2.indexer.indices.Indices.optimizeIndex(Indices.java:326) ~[graylog.jar:?]
        at org.graylog2.indexer.indices.jobs.OptimizeIndexJob.execute(OptimizeIndexJob.java:71) ~[graylog.jar:?]
        at org.graylog2.system.jobs.SystemJobManager$1.run(SystemJobManager.java:89) [graylog.jar:?]
        at com.codahale.metrics.InstrumentedScheduledExecutorService$InstrumentedRunnable.run(InstrumentedScheduledExecutorService.java:241) [graylog.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_282]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_282]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_282]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_282]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_282]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_282]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_282]
Caused by: java.net.SocketTimeoutException: 60,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
        at org.graylog.shaded.elasticsearch7.org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:834) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.elasticsearch.client.RestClient.performRequest(RestClient.java:259) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.elasticsearch.client.RestClient.performRequest(RestClient.java:246) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1613) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1583) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1553) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.elasticsearch.client.IndicesClient.forcemerge(IndicesClient.java:897) ~[?:?]
        at org.graylog.storage.elasticsearch7.IndicesAdapterES7.lambda$optimizeIndex$22(IndicesAdapterES7.java:394) ~[?:?]
        at org.graylog.storage.elasticsearch7.ElasticsearchClient.execute(ElasticsearchClient.java:95) ~[?:?]
        ... 13 more
Caused by: java.net.SocketTimeoutException: 60,000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
        at org.graylog.shaded.elasticsearch7.org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:92) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:39) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:502) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104) ~[?:?]
        at org.graylog.shaded.elasticsearch7.org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591) ~[?:?]
        ... 1 more
```
Reading in the docs and the forums, I was under the impression that timeout for these optimize jobs is controlled via the `elasticsearch_index_optimization_timeout`, which is set to 1 hour (and not 60 seconds).

Here is an excerpt of my server.conf setup :
```
$ cat /etc/graylog/server/server.conf | grep elasticsearch_index_optimization_timeout
elasticsearch_index_optimization_timeout = 1h
```
I have also doubled `elasticsearch_socket_timeout` and `elasticsearch_request_timeout` from 1m to 2m, but the timeouts did not disappear and the OptimizeIndexJob are still failing after 120 seconds. 

Any ideas ?

--- a/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java
+++ b/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java
@@ -16,6 +16,7 @@
  */
 package org.graylog.storage.elasticsearch7;
 
+import com.github.joschi.jadconfig.util.Duration;
 import com.google.common.collect.Streams;
 import org.graylog.shaded.elasticsearch7.org.apache.http.client.config.RequestConfig;
 import org.graylog.shaded.elasticsearch7.org.elasticsearch.ElasticsearchException;
@@ -31,7 +32,6 @@ import org.graylog2.indexer.InvalidWriteTargetException;
 import javax.inject.Inject;
 import javax.inject.Named;
 import java.io.IOException;
-import java.time.Duration;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
@@ -149,11 +149,11 @@ public class ElasticsearchClient {
         return e.getMessage().contains("index_not_found_exception");
     }
 
-    public static RequestOptions withTimeout(RequestOptions requestOptions, com.github.joschi.jadconfig.util.Duration timeout) {
+    public static RequestOptions withTimeout(RequestOptions requestOptions, Duration timeout) {
         final RequestConfig requestConfigWithTimeout = RequestConfig.copy(requestOptions.getRequestConfig())
                 .setSocketTimeout(Math.toIntExact(timeout.toMilliseconds()))
                 .build();
-        final RequestOptions requestOptionsWithTimeout = requestOptions.toBuilder()
+        return requestOptions.toBuilder()
                 .setRequestConfig(requestConfigWithTimeout)
                 .build();
 

--- a/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java
+++ b/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java
@@ -17,6 +17,7 @@
 package org.graylog.storage.elasticsearch7;
 
 import com.google.common.collect.Streams;
+import org.graylog.shaded.elasticsearch7.org.apache.http.client.config.RequestConfig;
 import org.graylog.shaded.elasticsearch7.org.elasticsearch.ElasticsearchException;
 import org.graylog.shaded.elasticsearch7.org.elasticsearch.action.search.MultiSearchRequest;
 import org.graylog.shaded.elasticsearch7.org.elasticsearch.action.search.MultiSearchResponse;
@@ -30,6 +31,7 @@ import org.graylog2.indexer.InvalidWriteTargetException;
 import javax.inject.Inject;
 import javax.inject.Named;
 import java.io.IOException;
+import java.time.Duration;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
@@ -145,5 +147,15 @@ public class ElasticsearchClient {
 
     private boolean isIndexNotFoundException(ElasticsearchException e) {
         return e.getMessage().contains("index_not_found_exception");
+    }
+
+    public static RequestOptions withTimeout(RequestOptions requestOptions, com.github.joschi.jadconfig.util.Duration timeout) {
+        final RequestConfig requestConfigWithTimeout = RequestConfig.copy(requestOptions.getRequestConfig())
+                .setSocketTimeout(Math.toIntExact(timeout.toMilliseconds()))
+                .build();
+        final RequestOptions requestOptionsWithTimeout = requestOptions.toBuilder()
+                .setRequestConfig(requestConfigWithTimeout)
+                .build();
+
     }
 }

--- a/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/IndicesAdapterES7.java
+++ b/graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/IndicesAdapterES7.java
@@ -87,6 +87,7 @@ import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
 import static java.util.stream.Collectors.toList;
+import static org.graylog.storage.elasticsearch7.ElasticsearchClient.withTimeout;
 
 public class IndicesAdapterES7 implements IndicesAdapter {
     private static final Logger LOG = LoggerFactory.getLogger(IndicesAdapterES7.class);
@@ -392,7 +393,7 @@ public class IndicesAdapterES7 implements IndicesAdapter {
                 .maxNumSegments(maxNumSegments)
                 .flush(true);
 
-        client.execute((c, requestOptions) -> c.indices().forcemerge(request, requestOptions));
+        client.execute((c, requestOptions) -> c.indices().forcemerge(request, withTimeout(requestOptions, timeout)));
     }
 
     @Override

--- a/graylog2-server/src/main/java/org/graylog2/configuration/DurationCastedToIntegerValidator.java
+++ b/graylog2-server/src/main/java/org/graylog2/configuration/DurationCastedToIntegerValidator.java
@@ -1,0 +1,40 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.configuration;
+
+import com.github.joschi.jadconfig.ValidationException;
+import com.github.joschi.jadconfig.Validator;
+import com.github.joschi.jadconfig.util.Duration;
+
+public class DurationCastedToIntegerValidator implements Validator<Duration> {
+    /**
+     * Validates if the value {@literal value} of the provided configuration parameter {@literal name} is a positive
+     * {@link Duration}, that is not higher than 24 days. This constraint is related to the duration (in milliseconds)
+     * being casted to integer at some point. {@link Integer#MAX_VALUE} / (24 * 60 * 60 * 1000) roughly equals to 24 days.
+     *
+     * @param name  The name of the configuration parameter
+     * @param value The value of the configuration validator
+     * @throws ValidationException If the value {@literal value} configuration parameter {@literal name} can't be parsed
+     *                             as a {@link Duration}, is negative or higher than 24 days.
+     */
+    @Override
+    public void validate(String name, Duration value) throws ValidationException {
+        if (value != null && (value.getQuantity() < 0L || value.toMilliseconds() > Integer.MAX_VALUE)) {
+            throw new ValidationException("Parameter " + name + " should be positive and not higher than 24 days (found " + value + ")");
+        }
+    }
+}

--- a/graylog2-server/src/main/java/org/graylog2/configuration/DurationCastedToIntegerValidator.java
+++ b/graylog2-server/src/main/java/org/graylog2/configuration/DurationCastedToIntegerValidator.java
@@ -21,6 +21,8 @@ import com.github.joschi.jadconfig.Validator;
 import com.github.joschi.jadconfig.util.Duration;
 
 public class DurationCastedToIntegerValidator implements Validator<Duration> {
+    private static final Duration maximumValue = Duration.days(24);
+
     /**
      * Validates if the value {@literal value} of the provided configuration parameter {@literal name} is a positive
      * {@link Duration}, that is not higher than 24 days. This constraint is related to the duration (in milliseconds)
@@ -33,7 +35,7 @@ public class DurationCastedToIntegerValidator implements Validator<Duration> {
      */
     @Override
     public void validate(String name, Duration value) throws ValidationException {
-        if (value != null && (value.getQuantity() < 0L || value.toMilliseconds() > Integer.MAX_VALUE)) {
+        if (value != null && (value.getQuantity() < 0L || value.toMilliseconds() > maximumValue.toMilliseconds())) {
             throw new ValidationException("Parameter " + name + " should be positive and not higher than 24 days (found " + value + ")");
         }
     }

--- a/graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java
+++ b/graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java
@@ -88,7 +88,7 @@ public class ElasticsearchConfiguration {
     @Parameter(value = "index_optimization_max_num_segments", validator = PositiveIntegerValidator.class)
     private int indexOptimizationMaxNumSegments = 1;
 
-    @Parameter(value = "elasticsearch_index_optimization_timeout", validator = PositiveDurationValidator.class)
+    @Parameter(value = "elasticsearch_index_optimization_timeout", validator = DurationCastedToIntegerValidator.class)
     private Duration indexOptimizationTimeout = Duration.hours(1L);
 
     @Parameter(value = "elasticsearch_index_optimization_jobs", validator = PositiveIntegerValidator.class)

